src/cowEndpointFunctions.py

```diff
@@ -0,0 +1,101 @@
+import json
+import requests
+from fractions import Fraction
+from config import ETHERSCAN_KEY
+
+
+def totalSurplus(startBlock = None, endBlock = None, settlementHash = None):
+    if settlementHash is not None:
+        getSurplus(settlementHash)
+
+    elif startBlock is not None and endBlock is not None:
+        #Etherscan endpoint call for settlements between start and end block
+        etherscanUrl = f"https://api.etherscan.io/api?module=account&action=txlist&address=0x9008D19f58AAbD9eD0D60971565AA8510560ab41&startblock={startBlock}&endblock={endBlock}&sort=desc&apikey={ETHERSCAN_KEY}"
+        jsonRecentSettlements = requests.get(etherscanUrl)
+        pyRecentSettlements = json.loads(jsonRecentSettlements.text)
+        #all "result" go into results (based on API return value names from docs)
+        results = pyRecentSettlements["result"]
+
+        for counterResults in range(len(results)):
+            settlementHash = results[counterResults]["hash"]
+            getSurplus(settlementHash)
+
+def fetchCompetitionData(settlementHash):
+    endpointUrl = f"https://api.cow.fi/mainnet/api/v1/solver_competition/by_tx_hash/{settlementHash}"
+    jsonCompetitionData = requests.get(endpointUrl)
+    return jsonCompetitionData
+
+            
+def getSurplus(settlementHash):   
+    # Once we have settlement transaction hashes, call competition endpoint to get solver data
+    jsonCompetitionData = fetchCompetitionData(settlementHash)
+    if jsonCompetitionData.status_code == 200:
```

Comment:
Can we write this in a `guard` form:

```
if jsonCompetitionData.status_code != 200 
   # print warning message
   continue

.. then the code for the case of success
```

It would be probably more readable.